llama-index-integrations/vector_stores/llama-index-vector-stores-azurecosmosnosql/llama_index/vector_stores/azurecosmosnosql/base.py

```diff
@@ -0,0 +1,276 @@
+                )
+
+        self._database_name = database_name
+        self._container_name = container_name
+        self._vector_embedding_policy = vector_embedding_policy
+        self._indexing_policy = indexing_policy
+        self._cosmos_container_properties = cosmos_container_properties
+        self._cosmos_database_properties = cosmos_database_properties
+        self._id_key = id_key
+        self._text_key = text_key
+        self._metadata_key = metadata_key
+        self._embedding_key = self._vector_embedding_policy["vectorEmbeddings"][0]["path"][1:]
+
+        self._database = self._cosmos_client.create_database_if_not_exists(
+            id=self._database_name,
+            offer_throughput=self._cosmos_database_properties.get("offer_throughput"),
+            session_token=self._cosmos_database_properties.get("session_token"),
+            initial_headers=self._cosmos_database_properties.get("initial_headers"),
+            etag=self._cosmos_database_properties.get("etag"),
+            match_condition=self._cosmos_database_properties.get("match_condition"),
+        )
+
+        # Create the collection if it already doesn't exist
+        self._container = self._database.create_container_if_not_exists(
+            id=self._container_name,
+            partition_key=self._cosmos_container_properties["partition_key"],
+            indexing_policy=self._indexing_policy,
+            default_ttl=self._cosmos_container_properties.get("default_ttl"),
+            offer_throughput=self._cosmos_container_properties.get("offer_throughput"),
+            unique_key_policy=self._cosmos_container_properties.get("unique_key_policy"),
+            conflict_resolution_policy=self._cosmos_container_properties.get("conflict_resolution_policy"),
+            analytical_storage_ttl=self._cosmos_container_properties.get("analytical_storage_ttl"),
+            computed_properties=self._cosmos_container_properties.get("computed_properties"),
+            etag=self._cosmos_container_properties.get("etag"),
+            match_condition=self._cosmos_container_properties.get("match_condition"),
+            session_token=self._cosmos_container_properties.get("session_token"),
+            initial_headers=self._cosmos_container_properties.get("initial_headers"),
+            vector_embedding_policy=self._vector_embedding_policy,
+        )
+
+    def add(
+            self,
+            nodes: List[BaseNode],
+            **add_kwargs: Any,
+    ) -> List[str]:
+        """Add nodes to index.
+
+        Args:
+            nodes: List[BaseNode]: list of nodes with embeddings
+
+        Returns:
+            A List of ids for successfully added nodes.
+
+        """
+        ids = []
+        data_to_insert = []
+
+        if not nodes:
+            raise Exception("Texts can not be null or empty")
+
+        #gets metadata from nodes
+        for node in nodes:
+            # print("This is node")
+            # print(node)
```

Comment:
leftover print statements - these are not the only ones, let's make sure to remove those altogether from the entire file